boa/src/syntax/lexer/mod.rs

```diff
@@ -252,6 +249,12 @@ impl<R> Lexer<R> {
                 '=' | '*' | '+' | '-' | '%' | '|' | '&' | '^' | '<' | '>' | '!' | '~' | '?' => {
                     Operator::new(next_ch as u8).lex(&mut self.cursor, start)
                 }
+                _ if c.is_digit(10) => {
+                    NumberLiteral::new(next_ch as u8).lex(&mut self.cursor, start)
+                }
+                _ if Identifier::is_identifier_start(c as u32) => {
```

Comment:
Why do we do the conversion to `u32` here?

Reply:
`c` here is a char, and the method `is_identifier_start` takes byte as input. I do this to make the match arms consistently processing only on `c`, but we could use `next_ch` here to prevent the type conversion.

Reply:
I think it's fine as-is :) thanks for the clarification!